tempodb/backend/cache/cache.go

```diff
@@ -4,25 +4,20 @@ import (
 	"context"
 	"io"
 
+	cortex_cache "github.com/cortexproject/cortex/pkg/chunk/cache"
 	"github.com/google/uuid"
 	"github.com/grafana/tempo/tempodb/backend"
 )
 
 type readerWriter struct {
 	nextReader backend.Reader
 	nextWriter backend.Writer
-	client     Client
+	cache      cortex_cache.Cache
 }
 
-type Client interface {
-	Fetch(ctx context.Context, key string) []byte
-	Store(ctx context.Context, key string, val []byte)
-	Shutdown()
-}
```

Comment:
Amazing work. Have been wanting to reuse the Cortex Cache interfaces for a long time!